ui/src/app/pages/signed-in/signed-in.tsx

```diff
@@ -0,0 +1,167 @@
+    height: '100%',
+    zIndex: -1,
+    position: 'absolute'
+  },
+});
+
+const checkOpsBeforeUnload = (e) => {
+  if (Object.keys(compoundRuntimeOpStore.get()).length > 0) {
+    // https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload
+    e.preventDefault();
+    e.returnValue = '';
+  }
+};
+
+interface Props extends WithSpinnerOverlayProps {
+  subscribeToInactivitySignOut: () => {};
+  signOut: () => {};
+}
+
+export const SignedIn = (props: Props) => {
+  useEffect(() => props.hideSpinner(), []);
+
+  const [hideFooter, setHideFooter] = useState(null);
+  const [subscriptions, setSubscriptions] = useState([]);
+
+  const {config} = useStore(serverConfigStore);
+  const {tiers} = useStore(cdrVersionStore);
+  const {profile} = useStore(profileStore);
+
+  useEffect(() => {
+    window.addEventListener('beforeunload', checkOpsBeforeUnload);
+
+    return () => {
+      window.removeEventListener('beforeunload', checkOpsBeforeUnload);
+    };
+  }, []);
+
+  useEffect(() => {
+    const subs = [];
+    subs.push(props.subscribeToInactivitySignOut());
+    // This handles detection of Angular-based routing data.
+    subs.push(routeConfigDataStore.subscribe(({minimizeChrome}) => {
+      setHideFooter(minimizeChrome);
+    }));
+    // This handles detection of React-based routing data. During migrations,
+    // we assume React routing data will be set deeper/later in the component
+    // hierarchy, therefore it will generally take precedence over React.
+    subs.push(routeDataStore.subscribe(({minimizeChrome}) => {
+      setHideFooter(minimizeChrome);
+    }));
+    setSubscriptions(subs);
+
+    return () => {
+      /*
+       * TODO RW-1920: we will only need one subscription after routeConfigDataStore and
+       *  routeDataStore are collapsed, and then we won't have to duck-type this
+       */
+      for (const s of subscriptions) {
+        s.unsubscribe();
+      }
+    };
+  }, []);
+
+  useEffect(() => {
+    /*
+     * TODO RW-6726 We want to block app rendering on the presence of server config data
+     *  so that route components don't try to lookup config data before it's available.
+     *  We should holistically figure out a Reacty way to do this.
+     *  See discussion on https://github.com/all-of-us/workbench/pull/4713
+     */
+    const checkStoresLoaded = async() => {
+      // AppComponent should be loading the server config.
+      if (config) {
+        if (!profile) {
+          profileStore.get().load();
+          return;
+        }
+        setInstitutionCategoryState(profile.verifiedInstitutionalAffiliation);
+        if (hasRegisteredAccess(profile.accessTierShortNames)) {
+          if (!tiers) {
+            const cdrVersionsByTier = await cdrVersionsApi().getCdrVersionsByTier();
+            cdrVersionStore.set(cdrVersionsByTier);
+            return;
+          }
+        }
+      }
+    };
+
+    checkStoresLoaded();
+  }, [profileStore, cdrVersionStore]);
```

Comment:
This is using `profileStore` and `cdrVersionStore` the *globals*. I'm not sure what the implications of this would be - `useEffect`'s second parameters are generally supposed to be props or state so that this gets evaluated at the same time when the render would be invoked again. Unfortunately the naming here is super overloaded but ideally I think this would look like:


```
const profileState = useStore(profileStore);
...
useEffect(() => {
  ...
}, [profileState, ...]);
```

It would also be ideal to invoke `load()` via the local store reference, e.g. `profileState.load()` rather than via the global. It might also work here if you  just reference `profile`.